src/main/java/datawave/data/type/TypeFactory.java

```diff
@@ -0,0 +1,44 @@
+package datawave.data.type;
+
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+/**
+ * TypeFactory that uses an internal loading cache to limit new Type objects
+ */
+public class TypeFactory {
+    
+    //  @formatter:off
+    private final LoadingCache<String, Type<?>> typeCache = CacheBuilder.newBuilder()
+                    .maximumSize(128)
```

Comment:
Any reason this should be configurable for size and expire?

Reply:
I am thinking the default expiration should be never.  128 as the default max size should account for all types thus far.  I agree that there should at least be methods to allow changing the maximumn size and expiration values.

Reply:
A config would alleviate this but I'm wondering if we should have any expiry at all on these objects. They are pretty lightweight and won't change so just limiting by the size makes a lot of sense to me.

Reply:
I'll add some default values for size and timeout. The cache code is a lift and shift from elsewhere in mainline datawave code and we often an expiration even for what should be short-lived code.